llama_index/query_engine/sub_question_query_engine.py

```diff
@@ -94,21 +94,18 @@ def from_defaults(
         elif len(query_engine_tools) > 0:
             callback_manager = query_engine_tools[0].query_engine.callback_manager
 
+        service_context = service_context or ServiceContext.from_defaults()
         if question_gen is None:
-            if service_context is None:
-                # use default openai model that supports function calling API
-                question_gen = OpenAIQuestionGenerator.from_defaults()
-            else:
-                # try to use OpenAI function calling based question generator.
-                # if incompatible, use general LLM question generator
-                try:
-                    question_gen = OpenAIQuestionGenerator.from_defaults(
-                        llm=service_context.llm
-                    )
-                except ValueError:
-                    question_gen = LLMQuestionGenerator.from_defaults(
-                        service_context=service_context
-                    )
+            # try to use OpenAI function calling based question generator.
+            # if incompatible, use general LLM question generator
+            try:
+                question_gen = OpenAIQuestionGenerator.from_defaults(
```

Comment:
OpenAIQuestionGenerator is kind of misleading since the underlying llm could be not openai.
could refactor it in future